### ISSUE TITLE
Jetty maven plugin - javax.annotation.jar picked up from jetty plugin rather than from applications classpath

### ISSUE THREAD

**Issue by thorodegaard** [eclipse/jetty.project#3186]
Application startup fails using maven jetty server.
Seems javax.annotation jar being picked up from jetty plugin rather than from applications classpath.

See attached stack trace and depencency tree for module with dependency to javax.annotation
[jetty_startup_java11.txt](https://github.com/eclipse/jetty.project/files/2648634/jetty_startup_java11.txt)
[dependency_tree_withannotations_dependency.txt](https://github.com/eclipse/jetty.project/files/2648635/dependency_tree_withannotations_dependency.txt)

**Comment by davidkarlsen** [comment]
Seems like a classloader leak?

**Comment by thorodegaard** [comment]
Configuration

`                    <groupId>org.eclipse.jetty</groupId>
                    <artifactId>jetty-maven-plugin</artifactId>
                    <configuration>
                        <useProvidedScope>true</useProvidedScope>
                        <useTestScope>true</useTestScope>
                        <webAppConfig>
                            <contextPath>/pin</contextPath>   
                        </webAppConfig>
                        <scanIntervalSeconds>10</scanIntervalSeconds>
                        <httpConnector>
                            <port>8080</port>
                        </httpConnector>
                        <systemPropertiesFile>src/test/resources/jettySystemProperties.properties</systemPropertiesFile>
                    </configuration>
`

**Comment by janbartel** [comment]
javax.annotation is indeed on the jetty plugin's classpath:  it is also on the container classpath (ie the equivalent) when you start jetty in the distribution. In general javax. classes are marked as system classes, meaning they are provided on the container classpath.

You could try adding the 1.3.2 version of the javax.annotation jar to the plugin dependencies and excluding it as a transitive dependency from your webapp's dependencies.

**Comment by davidkarlsen** [comment]
@janbartel We run on Java11 - where javax.annotations has been removed: https://blog.codefx.org/java/java-11-migration-guide/ - hence we need to add it as an explicit dependency. I can see that jetty uses this internally, but do not understand why it is required to leak into the webapp classloader?

**Comment by janbartel** [comment]
@davidkarlson all javax classes have been provided by container environment, and have been marked by the jetty webapp classloader as system classes (ie always loaded by the container classloader) for many years. Jetty has been providing the javax.annotations jar also for many years - if webapps have need of it at compile time then you can  add it as a provided dependency.

**Comment by davidkarlsen** [comment]
Hang on - when looking closer - the jar seems to be loaded from maven's lib dir - that's a leak - isn't it?

**Comment by olamy** [comment]
@davidkarlsen yup this has been recently added in maven core
any simple project do demonstrate the issue?

**Comment by davidkarlsen** [comment]
@olamy here you go https://github.com/davidkarlsen/jettybug

**Comment by davidkarlsen** [comment]
?
`
          <version>9.4.14.v20181114</version>
`

**Comment by janbartel** [comment]
Yes, but if I type "mvn jetty:run", it defaults to version 6.1.26.

**Comment by davidkarlsen** [comment]
```
Davids-Work-MacBook-Pro:jettybug et2448$ mvn jetty:run
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------------< test:test >------------------------------
[INFO] Building test 1.0-SNAPSHOT
[INFO] --------------------------------[ war ]---------------------------------
[INFO] 
[INFO] >>> jetty-maven-plugin:9.4.14.v20181114:run (default-cli) > test-compile @ test >>>
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ test ---
^Z
[1]+  Stopped                 mvn jetty:run
Davids-Work-MacBook-Pro:jettybug et2448$ 
```

**Comment by davidkarlsen** [comment]
If that doesn't work out for you, you can alway type:
`mvn org.eclipse.jetty:jetty-maven-plugin:9.4.14.v20181114:run`

**Comment by janbartel** [comment]
Ah! Something borked my local repo. Deleted it and now I can run. Thanks.

**Comment by janbartel** [comment]
@davidkarlsen If I exclude the jsr250-api jar from the classpath for the jetty:run-forked goal, I get the following error, could it be because the test harness isn't complete?:

> ***************************
> APPLICATION FAILED TO START
> ***************************
> Description:
> 
> A component required a bean named 'classpath:my.properties' that could not be found.
> 
> Action:
> 
> Consider defining a bean named 'classpath:my.properties' in your configuration.

**Comment by janbartel** [comment]
@davidkarlsen @thorodegaard  seems others have had this problem, and found a solution using standard maven configuration mechanism!

See https://stackoverflow.com/questions/52607814/how-to-remove-a-maven-lib-jsr250-api-1-0-jar-from-the-plugin-classpath-jetty

**Comment by olamy** [comment]
Solution overriding maven core dependency:
`
<build>
    <extensions>
      <extension>
        <groupId>javax.annotation</groupId>
        <artifactId>javax.annotation-api</artifactId>
        <version>${annotation-api.version}</version>
      </extension>
      <extension>
        <groupId>javax.annotation</groupId>
        <artifactId>jsr250-api</artifactId>
        <version>1.0</version>
      </extension>
    </extensions>
</build>
`

**Comment by olamy** [comment]
pr #3194 contains an it test with the workaround. @davidkarlsen thanks for your project I have used it with some modifications.
